core/core.go

```diff
@@ -237,27 +234,35 @@ func New(config Config) (*App, error) {
 		ethClient,
 		config.EthereumChainID,
 		config.EthereumRPCMaxContentLength,
-		config.OrderExpirationBuffer,
 	)
 	if err != nil {
 		return nil, err
 	}
 
+	latestHeader, err := meshDB.FindLatestMiniHeader()
+	if err != nil {
+		return nil, err
+	}
+	var latestBlockTimestamp *time.Time
+	if latestHeader != nil {
+		latestBlockTimestamp = &latestHeader.Timestamp
+	}
+
 	// Initialize order watcher (but don't start it yet).
 	orderWatcher, err := orderwatch.New(orderwatch.Config{
-		MeshDB:            meshDB,
-		BlockWatcher:      blockWatcher,
-		OrderValidator:    orderValidator,
-		ChainID:           config.EthereumChainID,
-		ExpirationBuffer:  config.OrderExpirationBuffer,
-		MaxOrders:         config.MaxOrdersInStorage,
-		MaxExpirationTime: metadata.MaxExpirationTime,
+		MeshDB:               meshDB,
+		BlockWatcher:         blockWatcher,
+		OrderValidator:       orderValidator,
+		ChainID:              config.EthereumChainID,
+		MaxOrders:            config.MaxOrdersInStorage,
+		MaxExpirationTime:    metadata.MaxExpirationTime,
+		LatestBlockTimestamp: latestBlockTimestamp,
```

Comment:
Does `LatestBlockTimestamp` really need to be passed in as part of `Config` on startup? It seems like:

1. Unless there was a sudden unexpected shutdown, we will have already pruned any expired orders when the block watcher notified us of this block before Mesh was shutdown.
2. The block watcher is going to notify us of the next latest block in a matter of seconds (or less) anyways.

Reply:
We could potentially miss a block re-org caused `Unexpired` event emission since on restart the OrderWatcher would not know what the `latestBlockTimestamp` was before the shutdown. It's an edge-case for sure, I just wanted to make sure the implementation was bullet proof. If you don't want to handle this edge-case, I can get rid of it.